ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -466,9 +466,17 @@ std::string filename_absolute(std::string fname) {
   if (file_exists(fname) || directory_exists(fname)) {
     wchar_t rpath[MAX_PATH];
     tstring tstr_fname = widen(fname);
-    GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL); 
-    return shorten(rpath);
-  } else { return ""; }
+    GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL);
+    if (directory_exists(ptr)) return add_slash(shorten(rpath));
```

Comment:
You're now checking for existence twice.

Reply:
While it isn't needed here, it is needed for realpath otherwise it would crash

Reply:
Anyway, I fixed it along with something you didn't point out that was also wrong with the code you referenced.